bugsnag-spring/build.gradle

```diff
@@ -1,52 +1,186 @@
 ext {
-    springVersion = '6.0.0'
-    springBootVersion = '3.0.0'
+    jakartaSpringVersion = '6.0.0'
+    jakartaSpringBootVersion = '3.0.0'
+
+    javaxSpringVersion = '5.3.20'
+    javaxSpringBootVersion = '2.5.14'
 }
 
 apply plugin: 'java-library'
 
-apply from: '../common.gradle'
-
 repositories {
     mavenCentral()
 }
 
-dependencies {
-    compileOnly project(':bugsnag-spring:javax')
-    api project(':bugsnag')
+java {
+    toolchain {
+        languageVersion.set(JavaLanguageVersion.of(8))
+    }
+}
 
-    compileOnly "jakarta.servlet:jakarta.servlet-api:${jakartaServletApiVersion}"
-    compileOnly "org.springframework:spring-webmvc:${springVersion}"
-    compileOnly "org.springframework.boot:spring-boot:${springBootVersion}"
-    compileOnly "ch.qos.logback:logback-core:${logbackVersion}"
-    compileOnly "org.slf4j:slf4j-api:${slf4jApiVersion}"
+// We use 3 custom configurations and matching sourceSets to avoid the standard behaviours which interfere with
+// our compilation structure
+configurations {
+    compileCommon
+    compileJavax
+    compileJakarta
+}
 
-    testImplementation project(':bugsnag').sourceSets.test.output
-    testImplementation project(':bugsnag-spring:javax')
-    testImplementation project(':bugsnag')
-    testImplementation "junit:junit:${junitVersion}"
-    testImplementation "jakarta.servlet:jakarta.servlet-api:${jakartaServletApiVersion}"
-    testImplementation "org.springframework.boot:spring-boot-starter-test:${springBootVersion}"
-    testImplementation "org.springframework.boot:spring-boot-starter-web:${springBootVersion}"
-    testImplementation "org.mockito:mockito-core:${mockitoVersion}"
+sourceSets {
+    common {
+        java
+        compileClasspath += configurations.compileCommon + configurations.compileJavax
```

Comment:
Is it correct that the common sourceSet has `compileJavax`?

Reply:
Yes. This allows the common classes to access the Spring classes that are not specific to either version. Unfortunately there isn't really a cleaner way to do this without complicating the dependencies of common and confusing things further.

Reply:
OK - suggest we add a comment here as a reminder to future readers.